include/fixed_containers/fixed_robinhood_hashtable.hpp

```diff
@@ -0,0 +1,574 @@
+    auto operator()(std::basic_string<CharT> const& str) const noexcept -> uint64_t {
+        return wyhash::hash(str.data(), sizeof(CharT) * str.size());
+    }
+};
+
+template <typename CharT>
+struct hash<std::basic_string_view<CharT>> {
+    auto operator()(std::basic_string_view<CharT> const& sv) const noexcept -> uint64_t {
+        return wyhash::hash(sv.data(), sizeof(CharT) * sv.size());
+    }
+};
+
+template <class T>
+struct hash<T*> {
+    auto operator()(T* ptr) const noexcept -> uint64_t {
+        return wyhash::hash(reinterpret_cast<uintptr_t>(ptr));
+    }
+};
+
+template <class T>
+struct hash<std::unique_ptr<T>> {
+    auto operator()(std::unique_ptr<T> const& ptr) const noexcept -> uint64_t {
+        return wyhash::hash(reinterpret_cast<uintptr_t>(ptr.get()));
+    }
+};
+
+template <class T>
+struct hash<std::shared_ptr<T>> {
+    auto operator()(std::shared_ptr<T> const& ptr) const noexcept -> uint64_t {
+        return wyhash::hash(reinterpret_cast<uintptr_t>(ptr.get()));
+    }
+};
+
+template <typename Enum>
+struct hash<Enum, typename std::enable_if<std::is_enum<Enum>::value>::type> {
+    auto operator()(Enum e) const noexcept -> uint64_t {
+        using underlying = typename std::underlying_type_t<Enum>;
+        return wyhash::hash(static_cast<underlying>(e));
+    }
+};
+
+#    define FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(T)                    \
+        template <>                                                      \
+        struct hash<T> {                                                 \
+            auto operator()(T const& obj) const noexcept -> uint64_t {   \
+                return wyhash::hash(static_cast<uint64_t>(obj)); \
+            }                                                            \
+        }
+
+#    if defined(__GNUC__) && !defined(__clang__)
+#        pragma GCC diagnostic push
+#        pragma GCC diagnostic ignored "-Wuseless-cast"
+#    endif
+// see https://en.cppreference.com/w/cpp/utility/hash
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(bool);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(char);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(signed char);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(unsigned char);
+#    if FIXED_ROBINHOOD_HASHTABLE_CPP_VERSION >= 202002L && defined(__cpp_char8_t)
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(char8_t);
+#    endif
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(char16_t);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(char32_t);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(wchar_t);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(short);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(unsigned short);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(int);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(unsigned int);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(long);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(long long);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(unsigned long);
+FIXED_ROBINHOOD_HASHTABLE_HASH_STATICCAST(unsigned long long);
```

Comment:
Should `#undef` the macro after use.
I would personally avoid using the macro altogether.
Seems this can be done with concepts?